roxctl/main.go

```diff
@@ -27,11 +31,41 @@ func main() {
 	// we simply add the usage information "(default false)" to our affected boolean flags.
 	AddMissingDefaultsToFlagUsage(c)
 
-	common.PatchPersistentPreRunHooks(c)
+	once := sync.Once{}
+	common.PatchPersistentPreRunHooks(c, func(cmd *cobra.Command, args []string) {
+		once.Do(func() {
```

Comment:
Could you please add a comment here on why this achieves what we want? IIUC we essentially perform a DFS on the command tree such that the function is first and only executed on the outer most child command.